compose/material/material/src/desktopMain/kotlin/androidx/compose/material/DesktopMenu.desktop.kt

```diff
@@ -530,91 +504,3 @@ fun Modifier.contextMenuOpenDetector(
         this
     }
 }
-
-/**
- * Positions a dropdown relative to another widget (its anchor).
- */
-@Immutable
-internal data class DesktopDropdownMenuPositionProvider(
```

Comment:
This class fixes the bug in this snippet:
```
import androidx.compose.foundation.layout.width
import androidx.compose.material.Button
import androidx.compose.material.DropdownMenu
import androidx.compose.material.DropdownMenuItem
import androidx.compose.material.Text
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import androidx.compose.ui.window.singleWindowApplication

fun main() = singleWindowApplication {
    var isMenuExpanded by remember { mutableStateOf(false) }

    Button(
        modifier = Modifier.width(160.dp),
        onClick = {
            isMenuExpanded = !isMenuExpanded
        }
    ) {
        Text("Click me")
    }
    DropdownMenu(
        modifier = Modifier.width(160.dp),
        expanded = isMenuExpanded,
        onDismissRequest = {
            isMenuExpanded = false
        },
    ) {
        repeat(10) {
            DropdownMenuItem(
                onClick = {
                    isMenuExpanded = false
                }
            ) {
                Text("Item $it")
            }
        }
    }
}
```

<img width="612" alt="image" src="https://github.com/JetBrains/compose-multiplatform-core/assets/5963351/a5a3dcca-4091-400d-b596-1aaa8c30ef56">

(menu opened above the button)

(reported [here](https://github.com/JetBrains/compose-multiplatform/issues/541#issuecomment-808789683) and [here](https://github.com/JetBrains/compose-multiplatform/issues/195))

We should move the fix to DropdownMenuPositionProvider if we remove DesktopDropdownMenuPositionProvider.